zebra-state/src/service/chain_tip.rs

```diff
@@ -320,6 +321,16 @@ impl ChainTipChange {
         Ok(action)
     }
 
+    /// Returns:
+    /// - `Some(TipAction)` if there has been a change since the last time the method was called.
+    /// - [`None`] if there has been no change.
+    pub fn get_tip_change(&mut self) -> Option<TipAction> {
```

Comment:
### Tricky shutdown edge case

If `tip_block_change` returns an error, that means the `watch::Sender` has been dropped, so we won't get any more state changes:
https://github.com/ZcashFoundation/zebra/blob/957e12e4ca28438ca2c64ca4a1870210f2b3e9b7/zebra-state/src/service/chain_tip.rs#L377-L380

I think we'll want to return that error here, so we can return an error from `Mempool::poll_ready`.
(This error should only happen when Zebra is shutting down.)

### Naming suggestion

Rust APIs typically don't use `get`: 
https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter

There might be a better name here, but I can't think of it right now. (Names are hard!)

### Documentation suggestion

It's good to avoid duplicating the detailed documentation on `wait_for_tip_change`.
But let's point to it, so people know where to look for details.

```suggestion
    ///
    /// See [`wait_for_tip_change`] for details.
    pub fn last_tip_change(&mut self) -> Result<Option<TipAction>, watch::error::RecvError> {
```

Reply:
Thanks for the warning about duplicating the docs. I was thinking it might not have been right what I did. Also, thanks for the note about `get`. Names consume a substantial amount of time in my workflow. What I sometimes do is that I name a particular thing `foo` and then rename it when the code is finished. This "lazy-evaluation" approach sometimes works.

Reply:
It's totally ok to take shortcuts to get something working, then ask the reviewer for help improving things.
(Some shortcuts are also totally ok to commit into the `main` branch.)

Reply:
Also I think what you've done with the documentation is fine here.

Reply:
Note: If we call `ChainTipChange::action` directly, the error here can't happen any more.

Reply:
> Note: If we call ChainTipChange::action directly, the error here can't happen any more.

I should have read the note more properly. I was still trying to return `Result` from `last_tip_change`, and I couldn't figure out how to return a `RecvError`. After a while, I realized I can just return `Option<TipAction>`.